### ISSUE TITLE
Provide a table for decision criterion for hashing to curve

### ISSUE THREAD

**Issue by ValarDragon** [arkworks-rs/crypto-primitives#16]
Currently we implement three different 'hash' functions for going from strings to elliptic curve elements. https://github.com/arkworks-rs/crypto-primitives/tree/main/src/crh

* Pedersen
* Bowe-Hopwood
* Injective map (I'm not really sure what this one is)

It'd probably be a handy reference to provide a table in the README for that directory explaining when to use each hash / what the different properties of each one is.

**Comment by Pratyush** [comment]
Yes, actually all of crypto-primitives is lacking documentation right now.
